Source/DataModels/Program/Program.h

```diff
@@ -75,34 +83,69 @@ inline void Program::BindUniformFloat4x4(const std::string& name, const float* d
 	glUniformMatrix4fv(glGetUniformLocation(id, name.c_str()), 1, transpose, data);
```

Comment:
Maybe this method can reuse the new one you added to avoid repeated code. I guess something like
```suggestion
	BindUniformFloat4x4(glGetUniformLocation(id, name.c_str()), data, transpose);
```
Same for the rest.

Reply:
i don't see the point to do it, if bindUniformFloat4x4 would be more extens i would do it, but calling a function just to do a line i don't like it.
PS: just because the compiler can ignore the inline (as can make function not inlined inline) in terms of what he thinks, the thing could change if we can ensure that inline works for gl functions (that now i'm not sure)

Reply:
Inline works for functions that call other functions. It's not inlining the entire second function iside the first one, just the redirection to its memory address. So shouldn't be an issue in that regard. Anyway, it's fine if you want to keep it like it is, it was just a suggestion, not a big deal.